backend/core/llm_chain.py

```
@@ -3,490 +3,320 @@
+        else:
+            del _retrieval_cache[cache_key]
+            logger.info(f"Stale retrieval cache entry removed: {cache_key}")
+    return None
 
-    except Exception as e:
-        logger.error(f"Error invoking chain: {e}")
-        raise
 
+def cache_retrieval(cache_key: str, documents: List[Document]):
+    """Stores a list of documents in the retrieval cache."""
+    if not cache_key or not ENABLE_CACHING: return
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)